Update library to support CX endpoint and follow Java API tags format

nazabic opened the pull request.

Commit:
Added customer experience reviews tag.

Commit:
Add customer experience reviews.

Commit:
Added specific tags for the CSS and Javascript assets.

Commit:
[+AD] Added special tags for conversations, and all the different types of badges. Also, the javascript tag has now been updated to support multiple trkrefs.

Commit:
Added Locale and NumberOfReviews attributes to the embedded content tags. Added tests and updated documentation.

plcstevens commented:
@nazabic so what tests are still missing?

nazabic commented:
The tests for ProductReviews, CustomerExperienceReviews and Conversations
classes.

On Wed, May 21, 2014 at 10:48 AM, Philip Stevens
notifications@github.comwrote:

> @nazabic https://github.com/nazabic so what tests are still missing?
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/reevoo/reevoomark-dotnet-api/pull/3#issuecomment-43735056
> .

Commit:
Write some more tests.

plcstevens commented on ReevooMark.Test/BadgesTests/CustomerServiceRatingBadgeTest.cs:
```diff
@@ -0,0 +1,50 @@
+			CustomerServiceRatingBadge cs_badge_without_trkref = new CustomerServiceRatingBadge ();
+			System.IO.StringWriter sw = new System.IO.StringWriter ();
+			HtmlTextWriter writer = new HtmlTextWriter(sw);
+			cs_badge_without_trkref.RenderControl(writer);
```
Dead test? Delete or re-implement.

plcstevens commented on ReevooMark.Test/BadgesTests/ProductBadgeTest.cs:
```diff
@@ -0,0 +1,51 @@
+			System.IO.StringWriter sw = new System.IO.StringWriter ();
+			HtmlTextWriter writer = new HtmlTextWriter(sw);
+			cs_badge_without_trkref.RenderControl(writer);
+			//	Console.WriteLine ("The TRKREF is: "+ cs_badge_without_trkref.TRKREF);
```
As above

plcstevens commented on ReevooMark.Test/ProductReviews.cs:
```diff
@@ -0,0 +1,12 @@
+	public class ProductReviews
+	{
+		public ProductReviews ()
+		{
```
No tests?

plcstevens commented on ReevooMark.Test/ReevooClientTests.cs:
```diff
@@ -67,9 +67,9 @@ private WebHeaderCollection CreateBadHeaders()
         {
             string _sku = @"foo";
-            string _tkref = @"bar";
+			string _trkref = @"bar";
```
I think you need to go through the project and make sure all tabs become spaces.

plcstevens commented on ReevooMark.Test/ReevooCustomerExperienceReviewsTest.cs:
```diff
@@ -0,0 +1,29 @@
+			HtmlTextWriter writer = new HtmlTextWriter(sw);
+			cx_reviews.SKU = "";
+			cx_reviews.TRKREF = "CYS";
+			cx_reviews.RenderControl (writer);
```
Delete

plcstevens commented on ReevooMark/Badges/AbstractReevooBadgeTag.cs:
```diff
@@ -0,0 +1,61 @@
+		public String BadgeClass {
+			get 
+			{
+				if (String.IsNullOrEmpty (VariantName))
```
I'd suggest always using brackets as it avoids issues where you refactor something in the future and your code goes crazy.

plcstevens commented on ReevooMark/JavascriptAssets.cs:
```diff
@@ -0,0 +1,59 @@
+	public class JavascriptAssets:AbstractReevooTag
+	{
+
+		const String SINGLE_TRKREF_MARKLOADER = "(function () {{var script = document.createElement('script'); " +
```
It may be worth googling how to have a constant equal to the contents of a file. That'd be better than large strings.

plcstevens commented:
If you could look to replace all tabs with spaces and delete the unneeded classes that'd be great.

Maybe worth a quick look to see if you can find a File loading way of setting constants, otherwise not a huge issue.

Need to also check why the tests are now failing.

Commit:
Added unit tests for delivery ratings badges and overall service rating badges. Also, cleaned and formatted the code to use 4 spaces instead of tabs.

Commit:
Cleaning and formatted the code.

Commit:
[+JL] *Added ability to display the tag body when the server returns a failure response code.
         *Added a new mock library that is compatible with .NET 2.0 and updated the unit tests
         *Updated the documentation

Commit:
[+JL] Update the unit tests to use properly the mocking framework.

rfonglip commented:
Any reason you using real client trkrefs? 

I should clarrify...in the documentation.

Commit:
Remove tests that require a network connection and that use reals trkrefs and skus.

Commit:
Refactoring the original "mark" tag to make sure that it has the same functionalities as before.

plcstevens commented:
Happy to merge once @nazabic has been able to upload the project to NuGet.

Commit:
Remove the new line characters from the javascript generated in order to make the test pass both on windows and unix.

Commit:
Solution and project files update.

Commit:
Removed some useless spaces from the string constant.
